### ISSUE TITLE
Conversion misses sections of Japanese Airspace

### ISSUE THREAD

**Issue by alus-** [alus-it/AirspaceConverter#78]
Bug reported by: John Blyth

Greetings from Japan!

I am currently towing gliders at Takikawa Skypark in Hokkaido, Japan.

I have been using your very nice Airspace Converter ((WIndows GUI version) to convert the OpenAIP data for Japan to Open Air, but I have found some sections are missing.

I don't know if it is important, but I can't use the unconverted AIP data in the Windows CE version of LK8000, but the Android version works fine.....

I am very happy to assist in tracking down this bug!

Cheers
John Blyth
(Usually in Carnarvon Western Australia)

For example airspace : "Takikawa GP 75FT"

I created a quick Excel spreadsheet to compare the original AIP file with the converted OpenAir one (generated by your program). It has a macro in it to compare the two data files.

There seem to be lots of missing defined places. Does your program have a limitation on the number of individual airspaces that it will convert?

It seems to skip everything after the "Js" or "Ks".....

I am happy to help in anyway I can.

[Missing Data in AIP to OpenAir Conversion.txt](https://github.com/alus-it/AirspaceConverter/files/2430754/Missing.Data.in.AIP.to.OpenAir.Conversion.txt)

[AIP COnversion Software - Data missing.xlsm.zip](https://github.com/alus-it/AirspaceConverter/files/2430756/AIP.COnversion.Software.-.Data.missing.xlsm.zip)

**Comment by alus-** [comment]
I found the bug!
When I wrote the part reading OpenAIP files I assumed (wrongly) that all airspaces in .aip have always correct altitude limits but in the Japanese file I discovered airspaces without valid limits (no references, for example: "Komatsu TCA-13") and in this case my converter is simply stopping and so ignoring all the rest.
Now I'm going to fix it and display warnings about the airspaces with wrong or inconsistent altitude limits (top and bottom at same altitude or swapped) and of course it will then continue to parse all the rest...

Stephan of OpenAIP may consider to verify and eventually suggest how to treat these cases.
